source/tyr/tyr/resources.py

```diff
@@ -2053,66 +2053,111 @@ def delete(self, id=None):
 
 
 class StreetNetworkBackend(flask_restful.Resource):
-    @marshal_with(streetnetwork_backend_list_fields)
-    def get(self, id=None):
-        if id:
-            try:
-                return {'streetnetwork_backends': [models.StreetNetworkBackend.find_by_id(id)]}
-            except sqlalchemy.orm.exc.NoResultFound:
-                return {'streetnetwork_backends': []}, 404
+    def _get_backend_by_id(self, backend_id):
+        return models.StreetNetworkBackend.query.get_or_404(backend_id)
+
+    def _get_all_backends(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument('page', type=int, required=False, default=1)
+        args = parser.parse_args()
+
+        pagination = models.StreetNetworkBackend.query.filter_by(discarded=False).paginate(
+            args['page'], current_app.config.get('MAX_ITEMS_PER_PAGE', 5)
+        )
+
+        pagination_json = {
+            'current_page': pagination.page,
+            'items_per_page': pagination.per_page,
+            'total_items': pagination.total,
+        }
+
+        if pagination.has_next:
+            pagination_json['next'] = url_for(request.endpoint, page=pagination.next_num)
+        return marshal(pagination.items, streetnetwork_backend_fields), pagination_json
+
+    def get(self, backend_id=None):
+        resp = None
+        if backend_id:
+            resp = marshal(self._get_backend_by_id(backend_id), streetnetwork_backend_fields)
         else:
-            return {'streetnetwork_backends': models.StreetNetworkBackend.all()}
+            resp = self._get_all_backends()
```

Comment:
Why losing the second return parameter (-> `pagination_json`) ?
Wouldn't it be better to call `marshal` at the same level for consistency ?

Reply:
`self._get_all_backends()` can return `pagination.items` and `pagination_json` and then will have something like that :
`backends_with_pagination = self._get_all_backends()`
`resp = marshal(backends_with_pagination[0], streetnetwork_backend_fields), backends_with_pagination[1]`
`return {'streetnetwork_backends': resp[0], 'pagination': resp[1]}`

The `if type(resp) == tuple` is useless here anyway, because `self._get_all_backends()` always returns a tuple.

Reply:
Sounds better to me !

Reply:
I'll adress this in a new PR after the release :)